#VRML_SIM R2022a utf8
WorldInfo {
}
Viewpoint {
  orientation -0.3026736074155525 0.46385567728991894 0.8326023048370491 1.2173529306914532
  position -1.836233117127912 -4.278918200674858 2.977651046177534
}
TexturedBackground {
}
RectangleArena {
  floorSize 5 5
}
Robot {
  translation -0.88 1.38 0.04
  rotation -1 1.88116e-06 9.58979e-09 -1.5708053071795867
  children [
    LightSensor {
      translation 0.029998 -9.23396e-07 0.0999999
      rotation -0.707104281180913 7.906683144102566e-07 -0.7071092811829013 3.14159
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ls_left"
    }
    LightSensor {
      translation -0.030002 -9.38564e-07 0.100001
      rotation 2.3464199999870813e-06 -0.9999999999944944 -2.3464199999870813e-06 1.57081
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ls_right"
    }
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 0.917647 0.145098 0.145098
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    DEF WHEEL1 HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.06 0 0.05
        rotation -5.339953139294501e-13 1 -5.339972754097044e-13 1.5707999999999998
        children [
          DEF WHEEL Transform {
            rotation 0 0 1 1.5708
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0.305882 0.898039 0.25098
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.02
                  radius 0.04
                  subdivision 24
                }
              }
            ]
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL2 HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation -0.06 0 0.05
        rotation -5.339200502354836e-13 1 -5.339220114392784e-13 1.5707999999999998
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL3 HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation 0.06 0 -0.05
        rotation 5.337337553878015e-13 1 5.337357159072958e-13 1.5707999999999998
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL4 HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.06 0 -0.05
        rotation 5.33738961945515e-13 1 5.337409224841341e-13 1.5707999999999998
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
  ]
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 1
  }
  controller "ls_controller"
}
PointLight {
  attenuation 0 0 1
  location 0 -1 0.1
}
